src/lib/portal-api.ts

```diff
@@ -250,3 +271,36 @@ function numericOrdinal(offering: IUserPortalOffering) {
   const ord = offering.problemOrdinal.split(".");
   return parseInt(ord[0], 10) * 1000 + parseInt(ord[1], 10);
 }
+
+export const getTeacherJWT = (domain: string, rawPortalJWT: string) => {
+  return new Promise<string>((resolve, reject) => {
+    superagent
+      .get(`${domain}api/v1/jwt/portal?as_teacher=true`)
+      .send({
+        as_teacher: true
+      })
+      .set("Authorization", `Bearer/JWT ${rawPortalJWT}`)
+      .end((err, res) => {
+        if (err) {
+          reject(getErrorMessage(err, res));
+        } else {
+          resolve(res.body.token);
+        }
+      });
+  });
+};
+
+export const getLearnerJWT = (domain: string, rawPortalJWT: string, offeringId: number) => {
+  return new Promise<string>((resolve, reject) => {
+    superagent
+      .get(`${domain}api/v1/jwt/portal?as_learner=true&offering_id=${offeringId}`)
+      .set("Authorization", `Bearer/JWT ${rawPortalJWT}`)
+      .end((err, res) => {
+        if (err) {
+          reject(getErrorMessage(err, res));
+        } else {
+          resolve(res.body.token);
+        }
+      });
+  });
```

Comment:
WIshlist: at some point we ought to centralize all of the similar code into some sort of a `make_portal_api_request` helper.